### DISCUSSION TITLE
Same AttributeError message in some getter function

### DISCUSSION THREAD

**Discussion by shenxiangzhuang** [georgebv/pyextremes#62] [Q&A] [answered]
I see many getter methods raise same `AttributeError` as fellow([code](https://github.com/georgebv/pyextremes/blob/e1e8ccdf7dfc5095e185bed0378cfa2d79192f67/src/pyextremes/eva.py#LL165C1-L165C34)):
```python
@property
def extremes_method(self) -> typing.Literal["BM", "POT"]:
    if self.__extremes_method is None:
        raise AttributeError(
            "extreme values must first be extracted "
            "using the '.get_extremes' method"
        )
    return self.__extremes_method
```
All of them with same error message `"extreme values must first be extracted using the '.get_extremes' method"`. Does this behavior is designed by some purpose  or just a wrong copy :-)

**Comment by georgebv** [comment, marked as answer]
The idea is that those attributes should not exist until user calls `get_extremes`. I had a choice between returning `None` and throwing an exception. Message is indeed the same - could be refactored to use global variable.
